qep-345-line-symbollayer-blankareas.md

```diff
@@ -0,0 +1,117 @@
+# QGIS Enhancement: Markers/Hashes line symbol layer blank areas 
+
+**Date** 2025/07/22
+
+**Author** Julien Cabieces (@troopa81)
+
+**Contact** julien dot cabieces at oslandia dot com
+
+**Version** QGIS 4.0
+
+# Summary
+
+This proposal relates to the line symbol layer which draw markers or hashes at intervals along a line. When using such symbol layer, the display of the repeated pattern (hash or marker) can be ugly, and user would like to have a better control on what's displayed.
+
+![issue.png](images/qep345/issue.png)
+
+The rest of the proposal will mention markers but everything describe here applies also to hashes.
+
+## Proposed Solution
+
+In order to have a proper display of the markers along the line, I propose to:
+- Give user a way to identify blank area where markers are not displayed
+- Use the already existing annotation feature to add, move, rotate or even change marker symbology
+
+This could look like approximatively this
+
+![poc.gif](images/qep345/poc.gif)
+
+### Fixed marker position
+
+We also propose to add an option "fixed marker position" to fix markers positions whatever the current displayed area and zoom level. Indeed, without this option, you would get this behavior
```

Comment:
I'm -1 to introducing a new option to control this. We already have the user-facing option for "Clip features to Canvas Extent" if this behaviour needs to be user controlled. If not, then I'd rather some heuristics are used to automatically set the the DisableFeatureClipping flag when appropriate (or just always set this for the marker/hash line types). Ideally accompanied with logic to skip the render of the markers/hashes outside the visible extent, to avoid performance issues.

Reply:
> Shall we really ? Isn't Qt already avoiding drawing anything if you draw outside the device (image, widget) ? Do we have something to gain to check beforehand ?

I'm not sure if Qt optimises and skips work for raster paint devices, but at least for svg/pdf devices it definitely includes objects which are outside of the device area.